api/disabled_parameters_test.go

```diff
@@ -14,6 +16,165 @@ import (
 	"github.com/algorand/indexer/api/generated/v2"
 )
 
+func TestToDisabledMapConfigFromFile(t *testing.T) {
+	expectedValue := &DisabledMapConfig{Data: map[string]map[string][]string{
+		"/sampleEndpoint": {http.MethodGet: {"p2"}},
+	}}
+
+	configFile := filepath.Join("test_resources", "mock_disabled_map_config.yaml")
+
+	// Nil pointer for openapi3.swagger because we don't want any validation
+	// to be run on the config (they are made up endpoints)
+	loadedConfig, err := MakeDisabledMapConfigFromFile(nil, configFile)
+	require.NoError(t, err)
+
+	require.True(t, reflect.DeepEqual(*expectedValue, *loadedConfig))
+
+}
+
+func TestToDisabledMapConfig(t *testing.T) {
+	type testingStruct struct {
+		name        string
+		ddm         *DisplayDisabledMap
+		dmc         *DisabledMapConfig
+		expectError bool
+	}
+
+	tests := []testingStruct{
+		{"test 1",
+			&DisplayDisabledMap{Data: map[string]map[string][]map[string]string{
+				"/sampleEndpoint": {
+					"required": {{"p1": "enabled"}, {"p2": "disabled"}},
+					"optional": {{"p3": "enabled"}},
+				}}},
+			&DisabledMapConfig{Data: map[string]map[string][]string{
+				"/sampleEndpoint": {http.MethodGet: {"p2"}},
+			}},
+
+			false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			// Nil pointer for openapi3.swagger because we don't want any validation
+			// to be run on the config
+			dmc, err := test.ddm.toDisabledMapConfig(nil)
+
+			if test.expectError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.True(t, reflect.DeepEqual(*dmc, *test.dmc))
+			}
+		})
+	}
+
+}
+
+func TestSchemaCheck(t *testing.T) {
+	type testingStruct struct {
+		name        string
+		ddm         *DisplayDisabledMap
+		expectError bool
```

Comment:
If you replace bool with a string, you can verify the error message also. Testing against a substring is sometimes convenient for making things more flexible, like `require.Contains(t, "p1, p3", err.Error())`. There are also some fancy things you can do with `require.ErrorIs` and `require.ErrorAs` to find/compare error types, but I don't think they would add much to this test.